test/tools/Modules/WebListener/WebListener.psm1

```diff
@@ -133,48 +137,53 @@ function Start-WebListener
         New-ServerCertificate -CertificatePath $serverPfxPath -Password $serverPfxPassword
         New-ClientCertificate -CertificatePath $Script:ClientPfxPath -Password $Script:ClientPfxPassword
 
-        $Job = Start-Job {
-            $path = Split-Path -parent (get-command WebListener).Path -Verbose
-            Push-Location $path -Verbose
-            'appDLL: {0}' -f $using:appDll
-            'serverPfxPath: {0}' -f $using:serverPfxPath
-            'serverPfxPassword: {0}' -f $using:serverPfxPassword
-            'HttpPort: {0}' -f $using:HttpPort
-            'Https: {0}' -f $using:HttpsPort
-            'Tls11Port: {0}' -f $using:Tls11Port
-            'TlsPort: {0}' -f $using:TlsPort
+        $oldASPEnvPreference = $env:ASPNETCORE_ENVIRONMENT
+
+        try {
             $env:ASPNETCORE_ENVIRONMENT = 'Development'
-            dotnet $using:appDll $using:serverPfxPath $using:serverPfxPassword $using:HttpPort $using:HttpsPort $using:Tls11Port $using:TlsPort
+
+            $params = @{
+                FilePath = $appExe
+                PassThru = $true
+                UseNewEnvironment = $IsLinux # start a non-keyboard input blocking process on linux.
+            }
+
+            $webListenerProcess = Start-Process @params -ArgumentList @(
+                $serverPfxPath
+                $serverPfxPassword
+                $HttpPort
+                $HttpsPort
+                $Tls11Port
+                $TlsPort
+            )
+        } catch {
+            # rethrow any exception
+            throw $_
+        } finally {
+            $env:ASPNETCORE_ENVIRONMENT = $oldASPEnvPreference
         }
+
         $Script:WebListener = [WebListener]@{
             HttpPort  = $HttpPort
             HttpsPort = $HttpsPort
             Tls11Port = $Tls11Port
             TlsPort   = $TlsPort
-            Job       = $Job
+            Process   = $webListenerProcess
         }
 
         # Count iterations of $sleepMilliseconds instead of using system time to work around possible CI VM sleep/delays
         $sleepCountRemaining = $initTimeoutSeconds * 1000 / $sleepMilliseconds
         do
         {
             Start-Sleep -Milliseconds $sleepMilliseconds
-            $initStatus = $Job.ChildJobs[0].Output | Out-String
-            $isRunning = $initStatus -match $initCompleteMessage
+            $isRunning = (Get-WebListener).Process -ne $null
```

Comment:
This check doesn't indicate the process is running. The process object will always be there once it's created, but it might have exited already.

Reply:
Updated.